packages/node_modules/@ciscospark/internal-plugin-user/src/user.js

```diff
@@ -68,9 +68,7 @@ const User = SparkPlugin.extend({
     })
       .then((res) => {
         this.spark.credentials.set({supertoken: res.body.tokenData});
-        if (res.body.hasPassword) {
-          this.hasPassword = true;
-        }
+        this.hasPassword = true;
```

Comment:
oops sorry - this line should actually be removed. 

`activate()` fetches the auth token so that the user will be able to set their password. when this function is called hasPassword should be false actually.
`setPassword()` should set `hasPassword` to true
`verify()` actually hits the /activate endpoint on atlas (yes confusing name) and this determines if the password was set